prefect_postgres/credentials.py

```diff
@@ -1,3 +1,73 @@
 """
 Module containing functionality for authenticating with PostgreSQL databases
 """
+import psycopg
+from typing import Any, Dict, Optional, Union
+from pydantic import SecretStr
+from prefect.blocks.core import Block
+from psycopg import connect
+
+
+class PostgresDatabaseCredentials(Block):
+    """
+    Block used to manage authentication with a PostgreSQL database.
+    Attributes:
+        database: The name of the database to use.
+        username: The user name used to authenticate.
+        password: The password used to authenticate.
+        host: The host address of the database.
+        port: The port to connect to the database.
+        connect_args: The options which will be passed directly to the
+           psycopg connect() method as additional keyword arguments.
+    Example:
+        Load stored database credentials:
+        ```python
+        from prefect_postgres import DatabaseCredentials
+        database_block = DatabaseCredentials.load("BLOCK_NAME")
+        ```
+    """
+
+    _block_type_name = "Postgres Database Credentials"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/7G8c2Zz4j0yyhXqQ44SI2A/eee9ba482dd6b61862b588b6fd28ad81/PostgreSQL-Logo.png?h=250"  # noqa
+
+    username: Optional[str] = Field(default=None, description="The name of the database to use.")
+    password: Optional[SecretStr] = Field(default=None, description="The user name used to authenticate.")
+    database: Optional[str] = Field(default=None, description="The password used to authenticate.")
+    host: Optional[str] = Field(default=None, description="The host address of the database.")
+    port: Optional[str] = Field(default=None, description="The port to connect to the database.")
+    connect_args: Optional[Dict[str, Any]] = Field(default=None, title="Additional Configuration", description="Additional configuration to use when creating a database connection.")
```

Comment:
Does it make sense for these to be `Optional`? `get_connection` appears to require all of them

Reply:
Good call. We aren't accepting a connection URL so they are all required except for `connect_args` which should default to an empty dictionary.
```suggestion
    username: str = Field(default=None, description="The name of the database to use.")
    password: SecretStr = Field(default=None, description="The user name used to authenticate.")
    database: str = Field(default=..., description="The password used to authenticate.")
    host: str = Field(default=..., description="The host address of the database.")
    port: str = Field(default=..., description="The port to connect to the database.")
    connect_args: Dict[str, Any] = Field(default_factory=dict, title="Additional Configuration", description="Additional configuration to use when creating a database connection.")
```